tools/wptrunner/wptrunner/executors/executorchrome.py

```diff
@@ -1,9 +1,122 @@
+
+here = os.path.join(os.path.split(__file__)[0])
+
+
+class ChromeDriverPrintProtocolPart(PrintProtocolPart):
+    def setup(self):
+        self.webdriver = self.parent.webdriver
+        self.runner_handle = None
+
+    def load_runner(self):
+        url = urljoin(self.parent.executor.server_url("http"), "/print_reftest_runner.html")
+        self.logger.debug("Loading %s" % url)
+        try:
+            self.webdriver.url = url
+        except Exception as e:
+            self.logger.critical(
+                "Loading initial page %s failed. Ensure that the "
+                "there are no other programs bound to this port and "
+                "that your firewall rules or network setup does not "
+                r"prevent access.\e%s" % (url, traceback.format_exc(e)))
+            raise
+        self.runner_handle = self.webdriver.window_handle
+
+    def render_as_pdf(self, width, height):
+        page_ranges = self.webdriver.execute_script(
+            """
+let elem = document.querySelector("meta[name=reftest-pages]");
+return elem ? elem.content : null;
+""")
+        margin = 0.5
+        body = {
+            "cmd": "Page.printToPDF",
+            "params": {
+                "paperWidth": width / 2.54,
+                "paperHeight": height / 2.54,
+                "marginLeft": margin,
+                "marginRight": margin,
+                "marginTop": margin,
+                "marginBottom": margin,
+                "shrinkToFit": False,
+                "printBackground": True,
+            }
+        }
+        return (self.webdriver.send_session_command("POST", "goog/cdp/execute", body=body)["data"],
+                page_ranges)
+
+    def pdf_to_png(self, pdf_base64, ranges):
+        handle = self.webdriver.window_handle
+        self.webdriver.window_handle = self.runner_handle
+        print(self.webdriver.url)
+        try:
+            rv = self.webdriver.execute_async_script("""
+let callback = arguments[arguments.length - 1];
+render('%s').then(result => callback(result))""" % pdf_base64)
+            page_numbers = get_pages(ranges, len(rv))
+            rv = [item for i, item in enumerate(rv) if i + 1 in page_numbers]
+            return rv
+        finally:
+            self.webdriver.window_handle = handle
 
 
 class ChromeDriverProtocol(WebDriverProtocol):
     server_cls = ChromeDriverServer
+    implements = WebDriverProtocol.implements + [ChromeDriverPrintProtocolPart]
+
+
+class ChromeDriverPrintRefTestExecutor(WebDriverRefTestExecutor):
+    protocol_cls = ChromeDriverProtocol
+
+    def setup(self, runner):
+        super(ChromeDriverPrintRefTestExecutor, self).setup(runner)
+        self.protocol.pdf_print.load_runner()
+        self.has_window = False
+        with open(os.path.join(here, "reftest.js")) as f:
+            self.script = f.read()
+
+    def screenshot(self, test, viewport_size, dpi):
+        # https://github.com/web-platform-tests/wpt/issues/7135
```

Comment:
Nit; https://github.com/web-platform-tests/wpt/issues/7140 is the dpi issue and even then we may want to track dpi support for print-reftests separately from reftests.